cubic-server/thread_pool/PriorityThreadPool.hpp

```diff
@@ -114,7 +114,7 @@ class PriorityThreadPool {
         {
             const std::lock_guard<std::mutex> _guard(_toolBox.queueProtection);
             jobId = ++_toolBox.totalJobsPushed;
-            _toolBox.jobQueue.emplace_back(jobId, priority, std::move(realJobList));
+            _toolBox.jobQueue.emplace_back(PriorityThreadPoolUtility::Job {jobId, priority, std::move(realJobList)});
```

Comment:
That removes the usefulness of emplace_back which can construct in-place, this has to copy everything over.